app/models/orchestration_stack.rb

```diff
@@ -47,11 +47,17 @@ def self.create_stack(orchestration_manager, stack_name, template, options = {})
     klass = orchestration_manager.class::OrchestrationStack
     ems_ref = klass.raw_create_stack(orchestration_manager, stack_name, template, options)
 
+    if options[:tenant_name]
+      tenant_class = orchestration_manager.class::CloudTenant
+      tenant = tenant_class.find_by(:name => options[:tenant_name], :ems_id => orchestration_manager.id)
+    end
```

Comment:
@bzwei You are always referencing `tenant` below but only setting it if passed a `options[:tenant_name]`.  (I know it works but looks odd.)  Also, I think you can just query the base CloudTenant class since you are passing the `:ems_id` which would restrict it to the proper instance, correct?

If we do not have any CloudTenants with name set to `nil` you could remove the condition and just do this:
```ruby
  tenant = CloudTenant.find_by(:name => options[:tenant_name], :ems_id => orchestration_manager.id)
```